### ISSUE TITLE
set_is_estimated still allowing estimates to exist in the system

### ISSUE THREAD

**Issue by Craig44** [NIWAFisheriesModelling/CASAL2#211]
I have attached a model, basically I have added an estimate transformation that if X1 + X2 = 1. Then you can estimate X1 only because X2 = 1 - X1.

So I have added a transformation called SumToOne, which basically takes two estimates turns one off and every iteration sets it to the equations above.

This does a -e fine. But when I print the estimates both X1 and X2 are printed even though one is not estimated. This is an issue for MCMC , because we expect only X1 when doing an mcmc but the MPD.cpp prints out both X1 and X2. For some reason in MPD.cpp the call ->GetIsEstimated() is still returning X2 even though in the SumToOne estimate transformation we are setting X2->set_estimated(false).

Any thoughts

[example.zip](https://github.com/NIWAFisheriesModelling/CASAL2/files/1717832/example.zip)

**Comment by Zaita** [comment]
Have a look at is_in_objective() instead of is_estimated()

**Comment by Craig44** [comment]
This has been addressed in the code, using `is_estimated` and `is_in_objective_function()`. Recently we have removed the subcommand `mcmc_fixed`. This was due to this issue. Going from -e to -m the code hasn't considered how an mcmc deals with a covariance that is based on `n` parameters but we fix some of them for mcmc. For this reason users will have to manully run an a -e run to get the fixed value for parameter they want to fix for MCMC, and then run a separate MCMC with that value fixed
